src/test/components/SampleTooltipContents.test.js

```diff
@@ -6,48 +6,121 @@
+const GRAPH_HEIGHT = 10;
+function getSamplesPixelPosition(
+  sampleIndex: IndexIntoSamplesTable,
+  samplePosition: 'before' | 'after' = 'before'
+): CssPixels {
+  const quarterSample = PIXELS_PER_SAMPLE / 4;
+  // Compute the pixel position of either first or the second part of the sample.
+  // "sampleIndex * PIXELS_PER_SAMPLE" corresponds to the center of the sample,
+  //  we need to +/- quarter sample to find these places in the sample:
+  //
+  //                       before     after
+  //                       50% CPU    100% CPU
+  //                           v         v
+  // +--------------------+--------------------+--------------------+
+  // |                    |          //////////|                    |
+  // +--------------------+////////////////////+--------------------+
+  const beforeOrAfter =
+    samplePosition === 'before' ? -quarterSample : +quarterSample;
+  return sampleIndex * PIXELS_PER_SAMPLE + beforeOrAfter;
+}
 
 describe('SampleTooltipContents', function() {
-  function setup(profile: Profile, hoveredSampleIndex: number) {
+  autoMockCanvasContext();
+  beforeEach(addRootOverlayElement);
+  afterEach(removeRootOverlayElement);
+
+  function getProfileWithCPU(
+    threadCPUDelta: Array<number | null>,
+    threadCPUDeltaUnit: ThreadCPUDeltaUnit
+  ): Profile {
+    const { profile } = getProfileFromTextSamples(`
+      A    A    A              A
+      B    B    B              B
+      Cjs  Cjs  H[cat:Layout]  H[cat:Layout]
+      D    F    I[cat:Idle]
+      Ejs  Ejs
+    `);
+    // Let's put some values for CPU usage.
+    addCpuUsageValues(profile, threadCPUDelta, threadCPUDeltaUnit);
+
+    return profile;
+  }
+
+  function setup(
+    profile: Profile,
+    hoveredSampleIndex: number,
+    hoveredSamplePosition: 'before' | 'after' = 'before'
+  ) {
     const store = storeWithProfile(profile);
-    const state = store.getState();
-    const threadSelectors = getThreadSelectorsFromThreadsKey(0);
-    const fullThread = threadSelectors.getRangeFilteredThread(state);
-    const categories = getCategories(state);
-    const sampleUnits = getSampleUnits(state);
-    const maxThreadCPUDelta = getMaxThreadCPUDelta(state);
-    const interval = getProfileInterval(state);
-
-    render(
+    const threadsKey = 0;
+
+    jest
+      .spyOn(HTMLElement.prototype, 'getBoundingClientRect')
+      .mockImplementation(() => getBoundingBox(GRAPH_WIDTH, GRAPH_HEIGHT));
+
+    const { container } = render(
       <Provider store={store}>
-        <SampleTooltipContents
-          sampleIndex={hoveredSampleIndex}
-          cpuRatioInTimeRange={null}
-          fullThread={fullThread}
-          categories={categories}
-          sampleUnits={sampleUnits}
-          maxThreadCPUDelta={maxThreadCPUDelta}
-          interval={interval}
+        <TimelineTrackThread
+          threadsKey={threadsKey}
+          trackType="expanded"
+          trackName="Test Track"
```

Comment:
We had to change it to TimelineTrackThread so activity graph can send information back to the tooltip.